tools/schema_loader.cc

```diff
@@ -649,6 +649,89 @@ schema_ptr do_load_schema_from_schema_tables(const db::config& dbcfg, std::files
     }
 }
 
+schema_ptr do_load_schema_from_sstable(const db::config& dbcfg, std::filesystem::path sstable_path, sstring keyspace, sstring table) {
+    if (keyspace.empty()) {
+        keyspace = "my_keyspace";
+    }
+    if (table.empty()) {
+        table = "my_table";
+    }
+
+    db::nop_large_data_handler large_data_handler;
+    gms::feature_service feature_service(gms::feature_config_from_db_config(dbcfg));
+    cache_tracker tracker;
+    sstables::directory_semaphore dir_sem(1);
+    sstables::sstables_manager sst_man("tools::load_schema_from_sstable", large_data_handler, dbcfg, feature_service, tracker,
+        memory::stats().total_memory(), dir_sem,
+        [host_id = locator::host_id::create_random_id()] { return host_id; });
+    auto close_sst_man = deferred_close(sst_man);
+
+    schema_ptr bootstrap_schema = schema_builder(keyspace, table).with_column("pk", int32_type, column_kind::partition_key).build();
+
+    const auto ed = sstables::parse_path(sstable_path, keyspace, table);
+    const auto dir_path = sstable_path.parent_path();
+    data_dictionary::storage_options local;
+    auto bootstrap_sst = sst_man.make_sstable(bootstrap_schema, dir_path.c_str(), local, ed.generation, sstables::sstable_state::normal, ed.version, ed.format);
+
+    bootstrap_sst->load_metadata().get();
+
+    const auto& serialization_header = bootstrap_sst->get_serialization_header();
```

Comment:
This will fail on per-ma sstables. Is the caller prepared for the exception? Perhaps we should catch it here (and return optional\<schema_ptr>).

Reply:
The caller is prepared for the exception. This is called from `try_load_schema_autodetect()`, which has a `try...catch` around each attempt to load the schema, falling over to the next method, until there is no more.